common/types.go

```diff
@@ -22,20 +21,32 @@ func NewBigInt(v int64) BigInt {
 }
 
 func (b BigInt) MarshalText() ([]byte, error) {
-	return []byte(b.String()), nil
+	return b.Int.MarshalText()
 }
 
 func (b *BigInt) UnmarshalText(text []byte) error {
 	return b.Int.UnmarshalText(text)
 }
 
 func (b BigInt) MarshalJSON() ([]byte, error) {
-	return []byte(fmt.Sprintf(`"%s"`, b.String())), nil
+	t, err := b.MarshalText()
+	if err != nil {
+		return nil, err
+	}
+	return json.Marshal(string(t))
```

Comment:
philosophically, we said "we want these to be strings in JSON." the issue with the previous code is that it further defines how strings are encoded in JSON. this updated, 5-times-as-many-lines code rather does less, allowing the json module to define how strings are encoded in json